includes/class-sensei-course.php

```diff
@@ -2641,50 +2641,44 @@ public static function archive_header( $query_type = '', $before_html = '', $aft
 	public static function single_course_content( $content ) {
 
 		if ( ! is_singular( 'course' ) ) {
-
 			return $content;
-
 		}
 
 		// Content Access Permissions
-		$access_permission = false;
-
-		if ( ! Sensei()->settings->get( 'access_permission' ) || sensei_all_access() ) {
-
-			$access_permission = true;
+		$is_login_required = true;
+		$has_full_access   = sensei_all_access();
 
+		if ( ! Sensei()->settings->get( 'access_permission' ) ) {
+			$is_login_required = false;
 		} // End If Statement
 
-		// Check if the user is taking the course
-		$is_user_taking_course = Sensei_Utils::user_started_course( get_the_ID(), get_current_user_id() );
-
-		if ( Sensei_WC::is_woocommerce_active() ) {
-
-			$wc_post_id = get_post_meta( get_the_ID(), '_course_woocommerce_product', true );
-			$product    = Sensei()->sensei_get_woocommerce_product_object( $wc_post_id );
-
-			$has_product_attached = isset( $product ) && is_object( $product );
-
-		} else {
-
-			$has_product_attached = false;
+		// By default, show content to admins (`$has_full_access`) and when login isn't required.
+		$has_access_to_content = ! $is_login_required || $has_full_access;
 
+		// Logged in visitors who are enrolled can see the course content.
+		$is_user_taking_course = false;
```

Comment:
It seems like this logic can be skipped if `$has_access_to_content` is already `true`.